### PR TITLE
fix: coupon value assertion

### PR DESCRIPTION
**Pull request by csandru-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/__tests__/support/types/types.ts [hunk 1]
```diff
@@ -5,7 +5,9 @@ export type Address = {
   state: string;
   country: string;
   zipcode: string;
+  zipCode: string;
```
Comment (FabianGerke) [inline apps/web/__tests__/support/types/types.ts]: Do we need both of them?

**Review by FabianGerke — COMMENTED** [review]
Looks good, just one question below.

File: apps/web/__tests__/support/types/types.ts (hunk 1 above)
Comment (csandru-plenty) [reply apps/web/__tests__/support/types/types.ts]: we need the added one, not sure about the other, let me check to see what I can find 
I'll remove it, if thats an option

File: apps/web/__tests__/support/pageObjects/CartPageObject.ts [hunk 2]
```diff
@@ -73,19 +71,20 @@ export class CartPageObject {
...
   removeCoupon() {
     cy.getByTestId('couponRemove').click();
     return this;
   }
 
-  orderSummayAfterCouponRemoved(total: string) {
+  orderSummayAfterCouponRemoved() {
     cy.getByTestId('coupon-label').should('not.exist');
-    cy.getByTestId('total').contains(total);
+    cy.getByTestId('total').invoke('text').should('have.length.gt', 0);
```
Comment (ksted) [inline apps/web/__tests__/support/pageObjects/CartPageObject.ts]: **question**: What do we gain from checking the text itself here? Compared to something like this:

```
orderSummayAfterCouponApplyed() {
    cy.getByTestId('coupon-label').should('be.visible');
    cy.getByTestId('coupon-value').should('be.visible');
  }

  removeCoupon() {
    cy.getByTestId('couponRemove').click();
    return this;
  }

  orderSummayAfterCouponRemoved() {
    cy.getByTestId('coupon-label').should('not.exist');
    cy.getByTestId('coupon-value').should('not.exist');
```

**suggestion**: I don't think this is necessarily the appropriate place to check the `total` element either. That one's part of the regular order summary, regardless of whether we apply a coupon or not. We might want to introduce a separate method instead, e.g. `cart.hasOrderSummary`, that checks for the order summary's structure as a whole.

File: apps/web/__tests__/test/smoke/cartPage.cy.ts [hunk 3]
```diff
@@ -1,28 +1,25 @@
...
-    cart.openCouponAccordion();
-    cart.addCoupon('KB82AZ');
-    cart.orderSummayAfterCouponApplyed('-£12.91', '£119.99');
-    cart.removeCoupon();
-    cart.orderSummayAfterCouponRemoved('£132.90');
+
+    cy.wait(1000)
+      .intercept('/plentysystems/doAddCartItem')
+      .as('doAddCartItem')
+      .wait(1000)
```
Comment (ksted) [inline apps/web/__tests__/test/smoke/cartPage.cy.ts]: **question**: Are you sure both `wait(1000)` statements are necessary? Cypress should retry for a few seconds by itself.

File: apps/web/__tests__/support/pageObjects/CartPageObject.ts [hunk 4]
```diff
@@ -71,20 +71,33 @@ export class CartPageObject {
...
+    cy.getByTestId('vat-label').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('vat').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('total-label').invoke('text').should('have.length.gt', 0);
     cy.getByTestId('total').invoke('text').should('have.length.gt', 0);
     return this;
   }
 
-  removeCoupon() {
-    cy.getByTestId('couponRemove').click();
+  orderSummaryAfterCouponApplyed() {
```
Comment (ksted) [inline apps/web/__tests__/support/pageObjects/CartPageObject.ts]:
```suggestion
  orderSummaryAfterCouponApplied() {
```

File: apps/web/__tests__/test/smoke/cartPage.cy.ts [hunk 5]
```diff
@@ -9,17 +9,17 @@ describe('Smoke: Cart Page', () => {
...
       .click()
       .wait('@doAddCartItem');
 
     cart
       .openCart()
+      .hasOrderSummary()
       .openCouponAccordion()
       .addCoupon('KB82AZ')
-      .orderSummayAfterCouponApplyed()
+      .orderSummaryAfterCouponApplyed()
```
Comment (ksted) [inline apps/web/__tests__/test/smoke/cartPage.cy.ts]:
```suggestion
      .orderSummaryAfterCouponApplied()
```

File: apps/web/__tests__/support/pageObjects/CartPageObject.ts [hunk 6]
```diff
@@ -71,20 +71,33 @@ export class CartPageObject {
...
+  }
+
+  hasOrderSummary() {
+    cy.getByTestId('subtotal-label').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('subtotal').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('shipping-label').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('shipping').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('vat-label').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('vat').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('total-label').invoke('text').should('have.length.gt', 0);
```
Comment (ksted) [inline apps/web/__tests__/support/pageObjects/CartPageObject.ts]:
```suggestion
    cy.getByTestId('subtotal-label').should('be.visible');
    cy.getByTestId('subtotal').should('be.visible');
    cy.getByTestId('shipping-label').should('be.visible');
    cy.getByTestId('shipping').should('be.visible');
    cy.getByTestId('vat-label').should('be.visible');
    cy.getByTestId('vat').should('be.visible');
    cy.getByTestId('total-label').should('be.visible');
```

**Review by ksted — CHANGES_REQUESTED** [review]

**Review by ksted — APPROVED** [review]

**Pull request merged**